src/caffe/layers/base_data_layer.cpp

```diff
@@ -77,7 +77,7 @@ void BasePrefetchingDataLayer<Dtype>::InternalThreadEntry() {
 #ifndef CPU_ONLY
   cudaStream_t stream;
   if (Caffe::mode() == Caffe::GPU) {
-    CUDA_CHECK(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
+    CUDA_CHECK(cudaStreamCreate(&stream));
```

Comment:
Non blocking streams are terrible for debugging and general sanity of the programmer. Non blocking was required because of the use of the default stream throughout caffe. Moving to the named stream(s) removes the need, so I had Dusan remove it. 

> On Aug 23, 2016, at 12:44 PM, Sergei Nikolaev <notifications@github.com> wrote:
> 
> In src/caffe/layers/base_data_layer.cpp:
> 
> > @@ -77,7 +77,7 @@ void BasePrefetchingDataLayer<Dtype>::InternalThreadEntry() {
> >  #ifndef CPU_ONLY
> >    cudaStream_t stream;
> >    if (Caffe::mode() == Caffe::GPU) {
> > -    CUDA_CHECK(cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking));
> > +    CUDA_CHECK(cudaStreamCreate(&stream));
> It came from Simon's PR.
> @thatguymike @slayton58 - guys, can you confirm that this is for good?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>